blocks/bricks/sequence_generators.py

```diff
@@ -714,8 +714,9 @@ def apply_delegate(self):
         # I can write self.apply because it can be overriden.
         # Thus I have to hack.
         # TODO: nice interface for this trick.
-        AttentionTransition.do_apply.__get__(self, None)
-        return AttentionTransition.do_apply
+        # Bart: This doesn't work anymore, and probably shouldn't; without a
+        # brick attached, it's impossible to call do_apply(brick)...
+        return self.do_apply
```

Comment:
Ehm, I am not sure I understand why `AttentionTransition.do_apply.__get__(self)` would not do the job. Doesn't it create a `BoundApplication` object similar to `self.do_apply`, with the exception of bypassing polymorphic method choice?

And by the way, I've just realized that we could in addition support `Brick.apply(instance)` interface, thus mimicking even closer behavior of usual methods. In order to do that the current `Application.__call__` should be renamed, and in fact there is no reason to call this function `__call__` currently. The new `__call__` method can create a bound application and pass the job the old `__call__` method.